src/buffer/out/Row.hpp

```diff
@@ -20,74 +20,139 @@ Revision History:
-    bool WasWrapForced() const noexcept { return _wrapForced; }
-
-    void SetDoubleBytePadded(const bool doubleBytePadded) noexcept { _doubleBytePadded = doubleBytePadded; }
-    bool WasDoubleBytePadded() const noexcept { return _doubleBytePadded; }
-
-    const CharRow& GetCharRow() const noexcept { return _charRow; }
-    CharRow& GetCharRow() noexcept { return _charRow; }
-
-    const ATTR_ROW& GetAttrRow() const noexcept { return _attrRow; }
-    ATTR_ROW& GetAttrRow() noexcept { return _attrRow; }
-
-    LineRendition GetLineRendition() const noexcept { return _lineRendition; }
-    void SetLineRendition(const LineRendition lineRendition) noexcept { _lineRendition = lineRendition; }
-
-    til::CoordType GetId() const noexcept { return _id; }
-    void SetId(const til::CoordType id) noexcept { _id = id; }
-
-    bool Reset(const TextAttribute Attr);
-    [[nodiscard]] HRESULT Resize(const til::CoordType width);
-
-    void ClearColumn(const til::CoordType column);
-    std::wstring GetText() const { return _charRow.GetText(); }
-
-    UnicodeStorage& GetUnicodeStorage() noexcept;
-    const UnicodeStorage& GetUnicodeStorage() const noexcept;
-
-    OutputCellIterator WriteCells(OutputCellIterator it, const til::CoordType index, const std::optional<bool> wrap = std::nullopt, std::optional<til::CoordType> limitRight = std::nullopt);
+    ROW() = default;
+    ROW(wchar_t* charsBuffer, uint16_t* charOffsetsBuffer, uint16_t rowWidth, const TextAttribute& fillAttribute);
+
+    ROW(const ROW& other) = delete;
+    ROW& operator=(const ROW& other) = delete;
+
+    explicit ROW(ROW&& other) = default;
+    ROW& operator=(ROW&& other) = default;
+
+    friend void swap(ROW& lhs, ROW& rhs) noexcept;
+
+    void SetWrapForced(const bool wrap) noexcept;
+    bool WasWrapForced() const noexcept;
+    void SetDoubleBytePadded(const bool doubleBytePadded) noexcept;
+    bool WasDoubleBytePadded() const noexcept;
+    void SetLineRendition(const LineRendition lineRendition) noexcept;
+    LineRendition GetLineRendition() const noexcept;
+
+    void Reset(const TextAttribute& attr);
+    void Resize(wchar_t* charsBuffer, uint16_t* charOffsetsBuffer, uint16_t rowWidth, const TextAttribute& fillAttribute);
+    void TransferAttributes(const til::small_rle<TextAttribute, uint16_t, 1>& attr, til::CoordType newWidth);
+
+    void ClearCell(til::CoordType column);
+    OutputCellIterator WriteCells(OutputCellIterator it, til::CoordType columnBegin, std::optional<bool> wrap = std::nullopt, std::optional<til::CoordType> limitRight = std::nullopt);
+    bool SetAttrToEnd(til::CoordType columnBegin, TextAttribute attr);
+    void ReplaceCharacters(til::CoordType columnBegin, til::CoordType width, const std::wstring_view& chars);
+
+    const til::small_rle<TextAttribute, uint16_t, 1>& Attributes() const noexcept;
+    TextAttribute GetAttrByColumn(til::CoordType column) const;
+    std::vector<uint16_t> GetHyperlinks() const;
+    uint16_t size() const noexcept;

```

Comment:
Makes it easier to work with `ROW`, since it stores `uint16_t` everywhere (to reduce the memory usage by a third). All integer types smaller than `int` get automatically converted to `int`, so even if we return `uint16_t` here, no warnings/errors/issues should pop up anywhere. But the reverse isn't true: Converting a `til::CoordType` to the `uint16_t` that `ROW` wants is annoying.